exceptions/app/src/main/cpp/native-lib.cpp

```diff
@@ -0,0 +1,15 @@
+#include <jni.h>
+
+#include <stdexcept>
+
+#include "exception_helper.h"
+
+extern "C" JNIEXPORT void JNICALL
+Java_com_example_exceptions_MainActivity_throwsException(JNIEnv* env,
+                                                         jobject /* this */) {
+  try {
+    throw std::runtime_error("A C++ runtime_error");
+  } catch (std::runtime_error e) {
+    jniThrowRuntimeException(env, e.what());
+  }
```

Comment:
It might be a better illustration here to move the throw to a function named something like `might_throw()`, and change the catch to just `std::exception`, since the goal is to prevent any exceptions from propagating across the JNI boundary (I suppose to be safe you should have the `std::exception` catch and also a `...` catch in case someone throws an `int` or whatever). A comment explaining that would also be good.

WDYT?

Reply:
PTAL